java-processors/src/main/java/com/dataiku/dip/plugins/tradearea/TradeAreaProcessor.java

```diff
@@ -0,0 +1,199 @@
+            return ProcessorDesc.withGenericForm(this.getName(), actionVerb("Create") + " trade area zone")
+                    .withMNEColParam("inputColumn", "Input column")
+                    .withColParam("outputColumn", "Output column")
+                    .withParam(ParamDesc.advancedSelect("unitMode", "Distance unit", "", UnitMode.class).withDefaultValue(UnitMode.KILOMETERS))
+                    .withParam(ParamDesc.advancedSelect("shapeMode", "Shape", "", ShapeMode.class).withDefaultValue(ShapeMode.CIRCLE))
+                    .withParam(ParamDesc.doubleP("radius", "Radius"))
+                    .withParam(ParamDesc.doubleP("width", "Width"))
+                    .withParam(ParamDesc.doubleP("height", "Height"));
+        }
+
+        @Override
+        public TradeAreaProcessor build(TradeAreaProcessor.Parameter parameter) throws Exception {
+            return new TradeAreaProcessor(parameter);
+        }
+    };
+
+    public TradeAreaProcessor(Parameter params) {
+        this.params = params;
+    }
+
+    private final Parameter params;
+    private Column outputColumn;
+    private Column cd;
+    private AreaGenerator generator;
+
+    @Override
+    public void processRow(Row row) throws Exception {
+        String str = row.get(cd);
+        // Handle null in input
+        if (str == null || str.equals("")) {
+            getProcessorOutput().emitRow(row);
+            return;
+        }
+
+        String output = "";
+        // Actual row to row processing
+        // str should be a WKT description of the input geospatial point
+        // example: POINT(-73.9723 40.64749) where -73.9723 is a longitude, 40.64749 is a latitude
+        GeoPoint.Coords coords = GeoPoint.convert(str);
+        if (coords != null){
+            output = generator.generateArea(coords);
+        }
+
+        // Handle null in output
+        if (output != null) {
+            row.put(outputColumn, output);
+        }
+
+        getProcessorOutput().emitRow(row);
+    }
+
+    @Override
+    public void postProcess() throws Exception {
+        getProcessorOutput().lastRowEmitted();
+    }
+
+    @Override
+    public void init() throws Exception {
+        cd = getCf().column(params.inputColumn, ProcessorRole.INPUT_COLUMN);
+        if (!StringUtils.isBlank(params.outputColumn)) {
+            outputColumn = getCf().column(params.outputColumn, ProcessorRole.OUTPUT_COLUMN);
+        } else {
+            outputColumn = cd;
+        }
+        generator = newGenerator(params.unitMode, params.shapeMode, params.radius, params.height, params.width);
+    }
+
+    @VisibleForTesting
+    static AreaGenerator newGenerator(UnitMode unitMode, ShapeMode shapeMode, double radius, double height, double width) {
+        logger.info("Instantiated new generator with argument: width="+width+" height="+height);
+        // Conversion factor from miles to km to enforce km as input
+        double milesToKm = 1.6093444978925633;
```

Comment:
declares it as a class constant 
```suggestion
      static final double MILES_TO_KM = 1.6093444978925633;
```

Reply:
Done